.github/workflows/release.yml

```diff
@@ -11,52 +11,16 @@ on:
         default: true
         description: "Dry run (doesn't actually publish anything)"
 
-env:
-  NUGET_SERVER_URL: https://www.nuget.org/
-  NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
-
 jobs:
 
   build:
     uses: ./.github/workflows/ci.yml
 
   deploy:
     needs: build
-    runs-on: ubuntu-22.04
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: true
-      - name: Download artifacts
-        uses: actions/download-artifact@v3
-        with:
-          name: windows-artifacts
-          path: /tmp/artifacts
-      - name: Deploy
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          if [[ "${{ github.event.inputs.dry-run }}" == "true" ]]; then
-            DRY_RUN_OPTION=--dry-run
-          fi
-          dotnet run --project tools-shared/FakeItEasy.Deploy -- $DRY_RUN_OPTION -r "${{ github.repository }}" -t "${{ github.ref_name }}" -a "/tmp/artifacts/output"
-
-  deploy-docs:
-    needs: deploy
-    runs-on: ubuntu-22.04
-    steps:
-      - name: Trigger docs generation
-        env:
-            GH_TOKEN: ${{ secrets.BOT_GITHUB_TOKEN }}
-        run: |
-          publishMode=auto-merge
-          if [[ "${{ github.event.inputs.dry-run }}" == "true" ]]; then
-            publishMode=none
-          fi
-          gh workflow run generate-docs.yml \
-            -R FakeItEasy/fakeiteasy.github.io \
-            -f output-prefix=docs/analyzers \
-            -f version=${{ github.ref_name }} \
-            -f source-repo=${{ github.repository }} \
-            -f source-ref=${{ github.ref_name }} \
-            -f publish-mode=$publishMode
+    uses: FakeItEasy/FakeItEasy.Tools/.github/workflows/deploy.yml@43ce145

```

Comment:
You have to embed the SHA?
I am shocked. I'd've assumed it would come from the current submodule version.
If we update tools, we need to change this here?
(supposing the workflow changes, I mean)

Reply:
> You have to embed the SHA?

Actually, it doesn't work with a short SHA1, it has to be the full one. It can also be a tag or branch. I think a branch is risky (we don't want to pick up changes until we're ready for them). We don't have tags in the FakeItEasy.Tools repo yet, but we could add them.

> I am shocked. I'd've assumed it would come from the current submodule version.

It doesn't rely on submodules at all. See my last messages in the chat

> If we update tools, we need to change this here?

Yes, if we want to use the updated workflow (or, we could just reference master, but as I said above, it's risky)

Reply:
Oh, right. owner/repo/…

I'm glad the short SHA doesn't work. What would happen when our repo got more commits and there were two that matched it all of a sudden?